apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/_components/EditSystemDialog.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import React, { useState } from "react";
+
+import * as schema from "@ctrlplane/db/schema";
+import { Button } from "@ctrlplane/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@ctrlplane/ui/dialog";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormRootError,
+  useForm,
+} from "@ctrlplane/ui/form";
+import { Input } from "@ctrlplane/ui/input";
+import { Textarea } from "@ctrlplane/ui/textarea";
+
+import { api } from "~/trpc/react";
+
+export const EditSystemDialog: React.FC<{
+  system: schema.System;
+  children: React.ReactNode;
+  onSubmit?: () => void;
+}> = ({ system, children, onSubmit }) => {
+  const [open, setOpen] = useState(false);
+
+  const form = useForm({
+    schema: schema.updateSystem,
+    defaultValues: {
+      name: system.name,
+      slug: system.slug,
+      description: system.description,
+    },
+  });
+
+  const updateSystem = api.system.update.useMutation();
+
+  const onFormSubmit = form.handleSubmit((data) => {
+    updateSystem.mutate(
+      { id: system.id, data },
+      {
+        onSuccess: () => {
```

Comment:
router refresh

Reply:
be consistent on your pattern. here you are using the hooks, other places you are using promises. Seems our code has defaulted to using promises